internal/handler/userHandler_test.go

```diff
@@ -0,0 +1,290 @@
+	mockService.AssertNumberOfCalls(t, "SaveUser", 0)
+	mockService.AssertNumberOfCalls(t, "GetUserById", 0)
+}
+
+func TestFindById(t *testing.T) {
+	gin.SetMode(gin.TestMode)
+	recorder := httptest.NewRecorder()
+	ctx, _ := gin.CreateTestContext(recorder)
+	mockService := mocks.NewService(t)
+	id := uuid.New().String()
+
+	pathParam := []gin.Param{
+		{
+			Key:   "id",
+			Value: id,
+		},
+	}
+	ctx.Params = pathParam
+
+	user, _ := models.NewUser(id, "John", "Doe", "johndoe@email.com", 18)
+	mockService.EXPECT().GetUserById(ctx, id).Return(&user, nil).Once()
+
+	handler := &Handler{
+		userService: mockService,
+	}
+
+	handler.FindById(ctx)
+	var responseBody response.UserResponseDto
+	_ = json.Unmarshal(recorder.Body.Bytes(), &responseBody)
+
+	assert.Equal(t, http.StatusOK, recorder.Code)
+	assert.Equal(t, user.GetID(), responseBody.ID)
+	assert.Equal(t, user.GetFirstName(), responseBody.FirstName)
+	assert.Equal(t, user.GetLastName(), responseBody.LastName)
+	assert.Equal(t, user.GetEmail(), responseBody.Email)
+	assert.Equal(t, user.GetAge(), responseBody.Age)
+	mockService.AssertExpectations(t)
+	mockService.AssertNumberOfCalls(t, "SaveUser", 0)
+}
+
+func TestUserDoesNotExist(t *testing.T) {
+	gin.SetMode(gin.TestMode)
+	recorder := httptest.NewRecorder()
+	ctx, _ := gin.CreateTestContext(recorder)
+	mockService := mocks.NewService(t)
+	id := uuid.New().String()
+
+	pathParam := []gin.Param{
+		{
+			Key:   "id",
+			Value: id,
+		},
+	}
+	ctx.Params = pathParam
+
+	mockService.EXPECT().GetUserById(ctx, id).Return(nil, nil).Once()
+
+	handler := &Handler{
+		userService: mockService,
+	}
+
+	handler.FindById(ctx)
+	var responseBody response.ErrorDto
+	_ = json.Unmarshal(recorder.Body.Bytes(), &responseBody)
+	errorMessage := "No user found with id " + id
+	assert.Equal(t, http.StatusNotFound, recorder.Code)
+	assert.Equal(t, errorMessage, responseBody.Message)
+	assert.NotEmpty(t, responseBody.Timestamp)
+	assert.Nil(t, responseBody.Field)
+	mockService.AssertExpectations(t)
+	mockService.AssertNumberOfCalls(t, "SaveUser", 0)
+}
+func TestErrorFindingUser(t *testing.T) {
+	gin.SetMode(gin.TestMode)
+	recorder := httptest.NewRecorder()
+	ctx, _ := gin.CreateTestContext(recorder)
+	mockService := mocks.NewService(t)
+	id := uuid.New().String()
+
+	pathParam := []gin.Param{
+		{
+			Key:   "id",
+			Value: id,
+		},
+	}
+	ctx.Params = pathParam
+
+	mockService.EXPECT().GetUserById(ctx, id).Return(nil, errors.New("some error")).Once()
+
+	handler := &Handler{
+		userService: mockService,
+	}
+
+	handler.FindById(ctx)
+	var responseBody response.ErrorDto
+	_ = json.Unmarshal(recorder.Body.Bytes(), &responseBody)
+	assert.Equal(t, http.StatusInternalServerError, recorder.Code)
+	assert.Equal(t, "Something went wrong", responseBody.Message)
+	assert.NotEmpty(t, responseBody.Timestamp)
+	assert.Nil(t, responseBody.Field)
+	mockService.AssertExpectations(t)
+	mockService.AssertNumberOfCalls(t, "SaveUser", 0)
+}
```

Comment:
These lines are not required, see comments in TestSaveUserSuccess.

Reply:
Fixed